### PR TITLE
Add truncated importance sampling and DrGRPO args

### PR DESCRIPTION
**Pull request by jacklanchantin**
**What does this PR do? Please describe:**

- [ ] Adds `tis_imp_ratio_cap` to use truncated importance sampling correction
- [ ] Adds GrpoLossConfig `adv_std_normarlization` (for DrGRPO)
- [ ] Adds new if statement to skip `ref_logps` computation for kl if beta == 0 (as done in DrGRPO)
- [ ] Adds  GrpoLossConfig `loss_token_mean` for normalizing over all tokens

Fixes #{issue number}

Most importantly, this adds truncated importance sampling correction, as recommended by @uralik.

**Does your PR introduce any breaking changes? If yes, please list them:**
List of all backwards-incompatible changes.

**Check list:**
- [ ] Was the content of this PR **discussed and approved** via a GitHub issue? (no need for typos or documentation improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairseq2/blob/main/CONTRIBUTING.md)?
- [ ] Did you make sure that your **PR does only one thing** instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**?
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you **update the [CHANGELOG](https://github.com/facebookresearch/fairseq2/blob/main/CHANGELOG.md)**? (no need for typos, documentation, or minor internal changes)

### CODE REVIEW

File: src/fairseq2/recipes/lm/_online_finetune/_grpo.py [hunk 1]
```diff
@@ -365,33 +380,49 @@ def _gather_lprobs(self, logits: Tensor, target: SequenceBatch) -> Tensor:
...
-        ]
-        # per_token_scaled_advantage = logps * advantages[:,:,None]
+        if self._config.loss_config.tis_imp_ratio_cap > 0:
+            tis_imp_ratio = torch.exp(model_logps - vllm_logps)
+            tis_imp_ratio = torch.clamp(
+                tis_imp_ratio, max=self._config.loss_config.tis_imp_ratio_cap
+            )
+            per_token_scaled_advantage = per_token_scaled_advantage * tis_imp_ratio
+
+        if ref_logps is not None:
```
Comment (jacklanchantin) [inline src/fairseq2/recipes/lm/_online_finetune/_grpo.py]: only use kl if `ref_logps` were computed

File: src/fairseq2/recipes/lm/_online_finetune/_grpo.py [hunk 2]
```diff
@@ -312,16 +319,21 @@ def __call__(
...
+        if self._config.loss_config.beta > 0:
+            ref_logps = compute_reference_logps(
+                self._gangs,
+                self._reference_model,
+                prompt_rollout_seqs,
+                prompt_rollout_layout,
+                grpo_batch.prompt_lengths,
+            )
+        else:
+            ref_logps = None
```
Comment (jacklanchantin) [inline src/fairseq2/recipes/lm/_online_finetune/_grpo.py:343]: skip this `ref_logps` computation if `beta == 0`

File: src/fairseq2/recipes/lm/_online_finetune/_grpo.py [hunk 3]
```diff
@@ -296,7 +300,8 @@ def __call__(
             grpo_input_batch_seqs, grpo_input_batch_seqs_layout
         )
 
-        logps = self._gather_lprobs(grpo_model_logits, grpo_target_batch)
+        model_logps = self._gather_lprobs(grpo_model_logits, grpo_target_batch)
+        vllm_logps = get_vllm_logprobs(rollouts[0], self._gangs).to(model_logps.device)
```
Comment (jacklanchantin) [inline src/fairseq2/recipes/lm/_online_finetune/_grpo.py]: `.to(model_logps.device)` works here, but i'm wondering if there's a better way to do this.

File: src/fairseq2/recipes/lm/_online_finetune/_grpo.py (hunk 1 above)
Comment (swarnaHub) [reply src/fairseq2/recipes/lm/_online_finetune/_grpo.py]: nitpick: this also means that beta is non-zero? does an assert make sense that it should never come here if beta is zero? or something that makes this if statement conditioned on beta for better readability?

otherwise LGTM!

**Review by uralik — CHANGES_REQUESTED** [review]
lets make sure this works for bs>1 before merging ! (as discussed offline)

File: src/fairseq2/recipes/lm/_online_finetune/_grpo.py [hunk 4]
```diff
@@ -296,7 +300,8 @@ def __call__(
             grpo_input_batch_seqs, grpo_input_batch_seqs_layout
         )
 
-        logps = self._gather_lprobs(grpo_model_logits, grpo_target_batch)
+        model_logps = self._gather_lprobs(grpo_model_logits, grpo_target_batch)
+        vllm_logps = get_vllm_logprobs(rollouts, self._gangs).to(model_logps.device)
```
Comment (jacklanchantin) [inline src/fairseq2/recipes/lm/_online_finetune/_grpo.py]: It works, but I'm not sure that setting `.to(model_logps.device)` is the right way to do this.
